doit/cmd_completion.py

```diff
@@ -129,15 +129,15 @@ def _zsh_arg_line(opt):
         # '(-c|--continue)'{-c,--continue}'[continue executing tasks...]' \
         # '--db-file[file used to save successful runs]' \
         if opt.short and opt.long:
-            tmpl = ("'(-{0.short}|--{0.long})'{{-{0.short},--{0.long}}}'"
-                    "[{help}]' \\")
+            tmpl = ('"(-{0.short}|--{0.long})"{{-{0.short},--{0.long}}}"'
+                    '[{help}]" \\')
         elif not opt.short and opt.long:
-            tmpl = "'--{0.long}[{help}]' \\"
+            tmpl = '"--{0.long}[{help}]" \\'
         elif opt.short and not opt.long:
-            tmpl = "'-{0.short}[{help}]' \\"
+            tmpl = '"-{0.short}[{help}]" \\'
         else: # without short or long options cant be really used
             return ''
-        ohelp = opt.help.replace(']', r'\]')
+        ohelp = opt.help.replace(']', r'\]').replace("'", r"\'")
```

Comment:
I don't understand... Since you modified the lines above replacing `'` with `"`, I guess here you should escape `"` not `'`.
I guess if you had modified only this line it would fix the problem.